pkg/minikube/config/config.go

```diff
@@ -74,3 +74,16 @@ func decode(r io.Reader) (MinikubeConfig, error) {
 	err := json.NewDecoder(r).Decode(&data)
 	return data, err
 }
+
+// MachineName is the name to use for the VM.
+const DefaultMachineName = "minikube"
+
+const MinikubeName = "MINIKUBE_NAME"
+
+// Minipath is the path to the user's minikube dir
+func GetMachineName() string {
+	if os.Getenv(MinikubeName) == "" {
```

Comment:
Why not just use `viper.GetString` here?  Viper looks at env vars, so you could call it `minikube-name` and that would translate to `MINIKUBE_NAME`

I think we should also add this to `minikube config set/unset`